src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,337 @@
+//! This module provides a view for breadcrumbs, enabling us to know which node the graph being
+//! edited belongs to and navigating through them.
+
+use crate::prelude::*;
+
+pub mod breadcrumb;
+pub mod project_name;
+
+pub use breadcrumb::Breadcrumb;
+pub use project_name::ProjectName;
+
+use enso_frp as frp;
+use ensogl::display;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use logger::enabled::Logger;
+use logger::AnyLogger;
+use ensogl::display::shape::text::text_field::FocusManager;
+use std::cmp;
+use enso_protocol::language_server::MethodPointer;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+// FIXME[dg] hardcoded literal for glyph of height 12.0. Copied from port.rs
+const GLYPH_WIDTH       : f32 = 7.224_609_4;
+const VERTICAL_MARGIN   : f32 = GLYPH_WIDTH;
+const HORIZONTAL_MARGIN : f32 = GLYPH_WIDTH;
+const TEXT_SIZE         : f32 = 12.0;
+
+
+
+// =================
+// === FrpInputs ===
+// =================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct FrpInputs {
+    /// Push breadcrumb.
+    pub push_breadcrumb : frp::Source<(Rc<MethodPointer>,uuid::Uuid)>,
+    /// Pop breadcrumb.
+    pub pop_breadcrumb : frp::Source,
+}
+
+impl FrpInputs {
+    /// Create new FrpInputs.
```

Comment:
I'd ask for one thing here. Let's try not to repeat names of structs in the doc. IT just makes refactoring harder. The doc "Constructor" would be as good as this one. If you really want to use name, always but it in backticks. Please fix in the whole pr.